src/mapping.ts

```diff
@@ -38,19 +41,19 @@ registerNodeFactory(ParserRuleContext, () => new GenericParseTreeNode());
 //Augment the ParseTree class with a toAST method
 declare module 'antlr4ts/tree' {
     export interface ParseTree {
-        toAST(parent?: Node): Node;
+        toAST(parent?: Node): Node | undefined;
     }
     export interface RuleNode {
-        toAST(parent?: Node): Node;
+        toAST(parent?: Node): Node | undefined;
     }
     export interface TerminalNode {
-        toAST(parent?: Node): Node;
+        toAST(parent?: Node): Node | undefined;
     }
 }
 
-RuleNode.prototype.toAST = function(parent?: Node): Node {
+RuleNode.prototype.toAST = function(parent?: Node): Node | undefined {
```

Comment:
...mentre qui possiamo essere certi che `toAST` non ritornerà mai undefined e quindi ritornare `toAST(this, parent)!`